app/src/main/java/ar/com/wolox/android/cookbook/twitterlogin/TwitterLoginRecipeFragment.kt

```diff
@@ -0,0 +1,121 @@
+package ar.com.wolox.android.cookbook.twitterlogin
+
+import android.content.Context
+import android.content.Intent
+import android.net.ConnectivityManager
+import android.net.NetworkInfo
+import android.net.Uri
+import android.view.View
+import android.widget.Toast
+import androidx.fragment.app.FragmentActivity
+import ar.com.wolox.android.cookbook.R
+import ar.com.wolox.android.cookbook.twitterlogin.model.YoutubeEmailResponse
+import ar.com.wolox.wolmo.core.fragment.WolmoFragment
+import com.twitter.sdk.android.core.Callback
+import com.twitter.sdk.android.core.TwitterSession
+import com.twitter.sdk.android.core.models.User
+import kotlinx.android.synthetic.main.fragment_twitter_login.*
+
+class TwitterLoginRecipeFragment : WolmoFragment<TwitterLoginRecipePresenter>(), TwitterLoginRecipeView {
+
+    override fun layout(): Int = R.layout.fragment_twitter_login
+
+    override fun init() {
+    }
+
+    override fun setListeners() {
+        vDefaultLoginBtn.setOnClickListener {
+            presenter.onDefaultBtnRequest()
```

Comment:
Remember, that Fragment is a silly class. I mean that it doesn't make decisions. So here methods should be called like
`onDefaultButtonClicked()`, `onCustomButtonClicked()`, etc. If you don't get it please ask me.

Reply:
i get it, i still had bad things from java development... my bad!